avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -57,4 +65,23 @@ protected PropertyName _findName(Annotated a)
         AvroName ann = _findAnnotation(a, AvroName.class);
         return (ann == null) ? null : PropertyName.construct(ann.value());
     }
+
+    @Override
+    public boolean hasCreatorAnnotation(Annotated a) {
+        AnnotatedConstructor constructor = a instanceof AnnotatedConstructor ? (AnnotatedConstructor) a : null;
+        AnnotatedClass parentClass =
+            a instanceof AnnotatedConstructor && ((AnnotatedConstructor) a).getTypeContext() instanceof AnnotatedClass
+            ? (AnnotatedClass) ((AnnotatedConstructor) a).getTypeContext()
+            : null;
+        return constructor != null && parentClass != null && parentClass.hasAnnotation(Stringable.class)
+            && constructor.getParameterCount() == 1 && String.class.equals(constructor.getRawParameterType(0));
+    }
+
+    @Override
+    public Object findSerializer(Annotated a) {
+        if (a instanceof AnnotatedClass && a.hasAnnotation(Stringable.class) || a.getRawType() == File.class) {
```

Comment:
Avro uses `toString()` / `getPath()` instead of `getAbsolutePath()`, so this was a quick fix to make the behavior match up. I'll replace it with a proper serializer in the module.